lua/lspconfig/configs.lua

```diff
@@ -37,13 +37,13 @@ function configs.__newindex(t, config_name, config_def)
       filetypes = { config.filetype, 't', true },
       on_new_config = { config.on_new_config, 'f', true },
       on_attach = { config.on_attach, 'f', true },
-      commands = { config.commands, 't', true },
+      user_commands = { config.user_commands, 't', true },
     }
-    if config.commands then
-      for k, v in pairs(config.commands) do
+    if config.user_commands then
+      for _, v in pairs(config.commands) do
```

Comment:
The last patch reverted this to `config.commands`, which should be `config.user_commands` ... and triggers errors as the check above doesn't work as intended.